modules/howtos/pages/subdocument-operations.adoc

```diff
@@ -261,50 +271,74 @@ Consider:
 Attempting to perform such an operation would result in a Path Not Found error.
 
 By default the automatic creation of parents is disabled, as a simple typo in application code can result in a rather confusing document structure.
 Sometimes it is necessary to have the server create the hierarchy however.
-In this case, the _create-path_ option may be used.
+In this case, the _create_path_ option may be used.
 
 [source,python]
 ----
 include::example$subdocument_ops.py[tag=createparents]
 ----
 
-== CAS Semantics
+== Concurrent Modifications
 
-Subdoc mostly eliminates the need for tracking the xref:concurrent-mutations-cluster.adoc[CAS] value.
-Subdoc operations are atomic and therefore if two different threads access two different sub-documents then no conflict will arise.
+Concurrent Sub-Document operations on different parts of a document will not conflict.
 For example the following two blocks can execute concurrently without any risk of conflict:
 
 [source,python]
 ----
 include::example$subdocument_ops.py[tag=cas1]
 ----
 
+Even when modifying the _same_ part of the document, operations will not necessarily conflict.
+For example, two concurrent _arrayappend_ operations to the same array will both succeed, never overwriting the other.
+
+
+So in some cases the application will not need to supply a xref:concurrent-mutations-cluster.adoc[CAS] value to protect
+against concurrent modifications.
+
+If CAS is required then it can be provided like this:
+
 [source,python]
 ----
 include::example$subdocument_ops.py[tag=cas2]
 ----
 
-Even when modifying the _same_ part of the document, operations will not necessarily conflict.
-For example, two concurrent _subdoc-array-append_ operations to the same array will both succeed, never overwriting the other.
 
-While CAS is no longer so strongly required to ensure document updates are preserved, as Sub-Doc reduces the chance of losing a mutation, it may still be needed to ensure document state remains consistent over multiple invocations of _mutate-in_: Sometimes it's important to ensure the entire document didn't change state since the last operation, such as in the case _subdoc-remove_ operations to ensure that the element being removed was not already replaced by something else.
 
-== Error handling
+== Durability
+
+Couchbase's xref:2.7@dotnet-sdk::durability.adoc[traditional durability], using `PersistTo` and `ReplicateTo`, is still available, particuarly for talking to Couchbase Server 6.0 and earlier:
 
-Subdoc operations have their own set of errors.
-When programming with subdoc, be prepared for any of the full-document errors (such as _Document Not Found_) as well as special sub-document errors which are received when certain constraints are not satisfied.
+[source,python]
+----
+include::example$subdocument_ops.py[tag=old_durability]
```

Comment:
This include and the next one render as empty atm:

<img src=https://user-images.githubusercontent.com/2589337/63025071-01710000-be98-11e9-90fd-5f7e2b8cd6ec.png width=300/>

Perhaps because the tag `old_durability` isn't in **subdocument_ops.py**?

Reply:
Correct. Will add a second PR to resolve this too.